pkgs/by-name/ge/gettext/package.nix

```diff
@@ -0,0 +1,125 @@
+}:
+
+# Note: this package is used for bootstrapping fetchurl, and thus cannot use
+# fetchpatch! All mutable patches (retrieved via by GitHub, cgit or similar
+# means) that perhaps are needed here should be included directly in Nixpkgs as
+# regular files inside this same directory.
+
+stdenv.mkDerivation (finalAttrs: ({
+  pname = "gettext";
+  version = "0.22.5";
+
+  src = fetchurl {
+    url = "mirror://gnu/gettext/gettext-${finalAttrs.version}.tar.gz";
+    hash = "sha256-7BcFselpuDqfBzFE7IBhUduIEn9eQP5alMtsj6SJlqA=";
+  };
+
+  patches = [
+    # Parameterize some tools to their absolute paths in gettext.sh.in
+    ./001-absolute-paths.diff
+    # fix reproducibile output, in particular in the grub2 build
+    # https://savannah.gnu.org/bugs/index.php?59658
+    ./002-msginit-do-not-use-pot-creation-date.patch
+  ]
+  ++ lib.optionals enableCoreServices [
+    # prevent infinite recursion for the darwin stdenv
+    ./003-revert-avoid-crash-on-macos-14.patch
+  ];
+
+  outputs = [ "out" "man" "doc" "info" ];
+
+  hardeningDisable = [ "format" ];
+
+  configureFlags = [
+    (lib.enableFeature false "csharp")
+    (lib.withFeature true "xz")
+  ] ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+    # On cross building, gettext supposes that the wchar.h from libc does not
+    # fulfill gettext needs, so it tries to work with its own wchar.h file,
+    # which does not cope well with the system's wchar.h and stddef.h (gcc-4.3 -
+    # glibc-2.9)
+    "gl_cv_func_wcwidth_works=yes"
+  ];
+
+  postPatch = ''
+   substituteAllInPlace gettext-runtime/src/gettext.sh.in
+   substituteInPlace gettext-tools/projects/KDE/trigger --replace "/bin/pwd" pwd
+   substituteInPlace gettext-tools/projects/GNOME/trigger --replace "/bin/pwd" pwd
+   substituteInPlace gettext-tools/src/project-id --replace "/bin/pwd" pwd
+  '' + lib.optionalString stdenv.hostPlatform.isCygwin ''
+    sed -i -e "s/\(cldr_plurals_LDADD = \)/\\1..\/gnulib-lib\/libxml_rpl.la /" gettext-tools/src/Makefile.in
+    sed -i -e "s/\(libgettextsrc_la_LDFLAGS = \)/\\1..\/gnulib-lib\/libxml_rpl.la /" gettext-tools/src/Makefile.in
+  '' + lib.optionalString stdenv.hostPlatform.isMinGW ''
+    sed -i "s/@GNULIB_CLOSE@/1/" */*/unistd.in.h
+  '';
+
+  nativeBuildInputs = [
+    xz
+    (lib.getBin xz)
+  ];
+
+  buildInputs = lib.optionals (!stdenv.hostPlatform.isMinGW) [
+    bash
+  ]
+  ++ lib.optionals enableLibiconv [
+    libiconv
+  ]
+  ++ lib.optionals enableCoreServices [
+    darwin.apple_sdk.frameworks.CoreServices
+  ];
+
+  strictDeps = true;
+
+  env = {
+    LDFLAGS = lib.optionalString stdenv.isSunOS "-lm -lmd -lmp -luutil -lnvpair -lnsl -lidmap -lavl -lsec";
+    gettextNeedsLdflags = stdenv.hostPlatform.libc != "glibc" && !stdenv.hostPlatform.isMusl;
+  };
+
+  enableParallelBuilding = true;
+  enableParallelChecking = false; # fails sometimes
+
+  setupHooks = [
+    ../../../build-support/setup-hooks/role.bash
```

Comment:
Looks like it won't pass the by-name check.

Reply:
I am cogitating  to write a dummy package that installs those scripts at Nix Store.